app/cni/cmd/install/main.go

```diff
@@ -0,0 +1,237 @@
+	}
+	log.V(1).Info("reverted config")
+	if err := removeKubeconfig(ic.MountedCniNetDir, ic.KubeconfigName); err != nil {
+		log.Error(err, "could not remove kubeconfig")
+	}
+	log.V(1).Info("removed kubeconfig")
+	log.Info("finished cleanup")
+}
+
+func removeKubeconfig(mountedCniNetDir, kubeconfigName string) error {
+	kubeconfigPath := mountedCniNetDir + "/" + kubeconfigName
+	if files.FileExists(kubeconfigPath) {
+		err := os.Remove(kubeconfigPath)
+		if err != nil {
+			return errors.Wrap(err, "couldn't remove kubeconfig file")
+		}
+	}
+	return nil
+}
+
+func revertConfig(mountedCniNetDir, cniConfName string, chainedCniPlugin bool) error {
+	configPath := mountedCniNetDir + "/" + cniConfName
+
+	if !files.FileExists(configPath) {
+		log.Info("no need to revert config - file does not exist", "configPath", configPath)
+		return nil
+	}
+
+	if chainedCniPlugin {
+		contents, err := ioutil.ReadFile(configPath)
+		if err != nil {
+			return errors.Wrap(err, "couldn't read cni conf file")
+		}
+		newContents, err := revertConfigContents(contents)
+		if err != nil {
+			return errors.Wrap(err, "could not revert config contents")
+		}
+		err = atomic.WriteFile(configPath, bytes.NewReader(newContents))
+		if err != nil {
+			return errors.Wrap(err, "could not write new conf")
+		}
+	} else {
+		err := os.Remove(configPath)
+		if err != nil {
+			return errors.Wrap(err, "couldn't remove cni conf file")
+		}
+	}
+
+	return nil
+}
+
+func install(ic *InstallerConfig) error {
+	if err := copyBinaries(); err != nil {
+		return err
+	}
+
+	if err := prepareKubeconfig(ic, serviceAccountPath); err != nil {
+		return err
+	}
+
+	if err := prepareKumaCniConfig(ic, serviceAccountPath); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func setupChainedPlugin(mountedCniNetDir, cniConfName, kumaCniConfig string) error {
+	resolvedName := cniConfName
+	extension := filepath.Ext(cniConfName)
+	if !files.FileExists(mountedCniNetDir+"/"+cniConfName) && extension == ".conf" && files.FileExists(mountedCniNetDir+"/"+cniConfName+"list") {
+		resolvedName = cniConfName + "list"
+	}
+
+	if files.FileExists(mountedCniNetDir + "/" + resolvedName) {
+		hostCniConfig, err := ioutil.ReadFile(mountedCniNetDir + "/" + resolvedName)
+		if err != nil {
+			return err
+		}
+
+		marshaled, err := transformJsonConfig(kumaCniConfig, hostCniConfig)
+		if err != nil {
+			return err
+		}
+		log.V(1).Info("resulting config", "config", string(marshaled))
+
+		err = atomic.WriteFile(mountedCniNetDir+"/"+resolvedName, bytes.NewReader(marshaled))
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+	return nil
+}
+
+func copyBinaries() error {
+	dirs := []string{primaryBinDir, secondaryBinDir}
+	writtenOnce := false
+	allErrors := errors.New("combined errors for copying binaries")
+	for _, dir := range dirs {
+		err := tryWritingToDir(dir)
+		if err != nil {
+			allErrors = errors.Wrapf(err, "could not write to dir %v", dir)
+			log.Error(err, "writing to dir failed", "dir", dir)
```

Comment:
should we log it? As far as I understand the logic, it's ok as long as we write to either primary or secondary dir. If we want to log it, we should probably log it on INFO, no?